src/plugins/intel_cpu/src/cpu_memory.cpp

```diff
@@ -42,7 +42,7 @@ namespace {
         if (!ftz) {
             return;
         }
-        if (src.getDesc().getPrecision() != ov::element::f32 || dst.getDesc().getPrecision() == ov::element::bf16) {
+        if (src.getDesc().getPrecision() != ov::element::f32 || dst.getDesc().getPrecision() != ov::element::f32) {
```

Comment:
@maxnick Please take a look. No sure about idea behind previous logic. Maybe you remember smt.
Anyway previous condition was incorrect for cases like src (fp32) -> dst (i32); it tried to apply ftz on i32 data.

Reply:
The original idea is to skip denormal nullifying for all the non float source types (as it doesn't make sense for integer types and bf16, as for the latter case HW implicitly assumes FTZ and DAZ flags are set when performing bf16 specific operations) also for bf16 dst type, as HW performs flushing denormal to zeros while converting float to bf16 using specific instructions.